cbcdb/main.py

```diff
@@ -0,0 +1,375 @@
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+            execute_flag: If set to false, the sql query will call the connection method to open tunnel and make the
+                          database connection. The connection method will recursively call this method with the
+                          execute_flag set to true.
+
+        Returns: A Pandas Dataframe.
+
+        """
+        self._print_debug_output(f"Getting query:\n {sql}")
+
+        if curs:
+            if params:
+                df = pd.read_sql_query(sql, params=params, con=conn)
+            else:
+                df = pd.read_sql_query(sql, con=conn)
+        else:
+            df = self._get_connection(sql, params, self.get_sql_dataframe)
+        return df
+
+    def get_sql_list_dicts(self, sql: str, params: list = None, curs=False, conn=False) -> List[Dict[str, Any]]:
+        """
+        Returns a list of dicts for a given SQL Query
+
+        Args:
+            sql: SQL string
+            params: Parameters for SQL call
+            curs: A cursor instance. If present the method will execute the SQL. If set to False, the method will
+            call the connection method. The connection method will recursively call this method with the cursor.
+
+
+        Returns:
+            Example Output:
+            [{'some': 'data'},
+             {'more': 'otherdata'}]
+
+        """
+        self._print_debug_output(f"Getting query:\n {sql}")
+        if curs:
+            if params:
+                curs.execute(sql, params)
+            else:
+                curs.execute(sql)
+            output = []
+            columns = [column[0] for column in curs.description]
+            for row in curs.fetchall():
+                output.append(dict(zip(columns, row)))
+        else:
+            output = self._get_connection(sql, params, self.get_sql_list_dicts)
+        return output
+
+    def get_sql_single_item_list(self, sql: str, params: list = None, curs=False, conn=False) -> list:
+        """
+        Returns a single column list for a given SQL Query
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+
+        Returns: A list containing the results of the query
+        """
+        self._print_debug_output(f"Getting query:\n {sql}")
+
+        if curs:
+            if params:
+                curs.execute(sql, params)
+            else:
+                curs.execute(sql)
+            output = []
+            for row in curs.fetchall():
+                output.append(row[0])
+        else:
+            output = self._get_connection(sql, params, self.get_sql_single_item_list)
+        return output
+
+    def execute_simple(self, sql: str, params: list = None, curs=False, conn=False):
+        """
+        Execute as single SQL statement
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+
+        Returns: None
+        """
+        if curs:
```

Comment:
In other places, I see the `curs.execute` method is called differently depending on the existence of `params`. I'd recommend extracting a method that executes a query given a cursor, sql statement, and optional params, that way there's less repeated code throughout the class, and the queries are made consistently